src/app/game-manager/game-manager.service.ts

```diff
@@ -0,0 +1,51 @@
+import { TimerService } from './../shared/timer/timer.service';
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject, BehaviorSubject } from 'rxjs';
+import { DataRepository } from '../repository/repository.service';
+import { Game } from '../shared/models/game.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GameManagerService {
+  private currentGame = new ReplaySubject<Game>();
+  private isGameStarted = new BehaviorSubject<boolean>(false);
+  private timerTimeout!: ReturnType<typeof setTimeout>;
+
+  constructor(
+    private repo: DataRepository,
+    private timerService: TimerService,
+  ) {}
+
+  get currentGameObservable(): Observable<Game> {
+    return this.currentGame.asObservable();
+  }
+
+  get gameStatus(): Observable<boolean> {
+    return this.isGameStarted.asObservable();
+  }
+
+  refreshCurrentGameInfo(): void {
+    // start the timer with calculated values;
+    this.repo.getCurrentGame().subscribe((game) => {
```

Comment:
what if there is no next game? or current game is finished? think about edge cases

Reply:
done